base/abstractarray.jl

```diff
@@ -1047,6 +1047,86 @@ function _setindex!(::IndexCartesian, A::AbstractArray, v, I::Vararg{Int,M}) whe
     r
 end
 
+"""
+    parent(A)
+
+Returns the "parent array" of an array view type (e.g., `SubArray`), or the array itself if
+it is not a view.
+
+# Examples
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> V = view(A, 1:2, :)
+2×2 view(::Array{Int64,2}, 1:2, :) with eltype Int64:
+ 1  2
+ 3  4
+
+julia> parent(V)
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+```
+"""
+parent(a::AbstractArray) = a
+
+## rudimentary aliasing detection ##
+"""
+    unalias(dest, A)
+
+Return either `A` or a copy of `A`, in a rough effort to prevent modifications to `dest` from
+affecting the returned object. No guarantees are provided, and each custom array must
+opt-into aliasing detection by overloading this method by specializing on the second argument.
+
+This function must return an object of exactly the same type as `A` for performance and type stability.
+
+See also [`mightalias`](@ref) and [`dataids`](@ref).
+"""
+unalias(dest, A) = mightalias(dest, A) ? copypreservingtype(A) : A
+
+copypreservingtype(A::Array) = copy(A)
+copypreservingtype(A::AbstractArray) = (@_noinline_meta; deepcopy(A)::typeof(A))
```

Comment:
I'm not sure `deepcopy` is ever correct. Maybe better just to give a method error?

Reply:
Ah, you're right, I didn't consider how this would change the identities of the values that get assigned into the array.  Perhaps we should just fall back to `copy(A)`.  It'll be type-unstable but this function exists to allow you to fix it.